pkg/daemon/route/route.go

```diff
@@ -165,33 +161,67 @@ func CreateRouteManager(localDirectTableNum, toOverlaySubnetTableNum, overlayMar
 		toOverlaySubnetTableNum: toOverlaySubnetTableNum,
 		overlayMarkTableNum:     overlayMarkTableNum,
 		family:                  family,
-		subnetInfoMap:           SubnetInfoMap{},
+		totalSubnetInfoMap:      SubnetInfoMap{},
+		overlaySubnetInfoMap:    SubnetInfoMap{},
+		underlaySubnetInfoMap:   SubnetInfoMap{},
 	}, nil
 }
 
 func (m *Manager) ResetInfos() {
-	m.subnetInfoMap = SubnetInfoMap{}
+	m.totalSubnetInfoMap = SubnetInfoMap{}
+	m.underlaySubnetInfoMap = SubnetInfoMap{}
+	m.overlaySubnetInfoMap = SubnetInfoMap{}
 }
 
-func (m *Manager) AddSubnetInfo(cidr *net.IPNet, gateway net.IP, forwardNodeIfName string, autoNatOutgoing, isOverlay bool) {
-	m.subnetInfoMap[cidr.String()] = &SubnetInfo{
-		cidr:              cidr,
-		forwardNodeIfName: forwardNodeIfName,
-		isOverlay:         isOverlay,
-		gateway:           gateway,
-		autoNatOutgoing:   autoNatOutgoing,
+func (m *Manager) AddSubnetInfo(cidr *net.IPNet, gateway, start, end net.IP, excludeIPs, reservedIPs []net.IP,
+	forwardNodeIfName string, autoNatOutgoing, isOverlay bool) {
+
+	cidrString := cidr.String()
+	if _, exist := m.totalSubnetInfoMap[cidrString]; !exist {
+		m.totalSubnetInfoMap[cidrString] = &SubnetInfo{
+			cidr:              cidr,
+			forwardNodeIfName: forwardNodeIfName,
+			gateway:           gateway,
+			autoNatOutgoing:   autoNatOutgoing,
+			includedIPRanges:  []*daemonutils.IPRange{},
+			excludeIPs:        []net.IP{},
+			reservedIPs:       []net.IP{},
+		}
+	}
+
+	subnetInfo := m.totalSubnetInfoMap[cidrString]
+
+	if len(reservedIPs) != 0 {
+		subnetInfo.reservedIPs = append(subnetInfo.reservedIPs, reservedIPs...)
+	}
+
+	if len(excludeIPs) != 0 {
+		subnetInfo.excludeIPs = append(subnetInfo.excludeIPs, excludeIPs...)
+	}
+
+	if start != nil || end != nil {
+		if start == nil {
+			start = cidr.IP
```

Comment:
actually we are using the second IP of network IP as start

Reply:
This is used for calculate a CIDR, not available ip to allocate, every possible ip should be considered.